### ISSUE TITLE
Can't add podcasts (EOF - forbidden)

### ISSUE THREAD

**Issue by kyrawertho** [akhilrex/podgrab#225]
Hi, I am just setting up Podgrab for the first time and I can't seem to add several podcasts. When I add them, an error message shows only "EOF" and Docker shows this:

/api/db/dbfunctions.go:17 record not found
[0.433ms] [rows:0] SELECT * FROM `podcasts` WHERE `podcasts`.`url` = "https://feeds.buzzsprout.com/1403161.rss" ORDER BY `podcasts`.`id` LIMIT 1

**Comment by akhilrex** [comment]
Let me see what I can do.

**Comment by jazzsnobeatcake** [comment]
I'm chiming in because I also face this issue and noticed that it's only happening when the feed URL points to feeds.buzzsprout.com, just like in @kyrawertho log. Podcasts hosted elsewhere are working fine for me.

**Comment by t3rminalx** [comment]
I've tried adding a podcast which also seems to come from buzzsprout and had the same EOF issue.

https://podcastaddict.com/podcast/4040551

**Comment by tomrossi7** [comment]
Hey guys! This is Tom from Buzzsprout. Are you able to set the user agent to something that reflects podgrab?

**Comment by wiseindy** [comment]
+1 I have the same issue with Buzzsprout podcasts: e.g.: https://podcastaddict.com/podcast/3386743

**Comment by tomrossi7** [comment]
Has the user agent been updated?

**Comment by wiseindy** [comment]
There's an open PR that fixes this issue: https://github.com/akhilrex/podgrab/pull/214

**Comment by t3rminalx** [comment]
> There's an open PR that fixes this issue: #214

Thats awesome, thank you. I've noticed you've approved the changes too. How long until we see the changes in the code?

**Comment by wiseindy** [comment]
Not sure, since I don't have the access to merge. Pinging @akhilrex

**Comment by wiseindy** [comment]
@tomrossi7 In the meantime, I created a fork here with the user agent fix: https://github.com/wiseindy/podgrab

If you use docker and want to use it, you can swap out the image `akhilrex/podgrab` with `wiseindy/podgrab` until we wait for the PR to be merged in this repo.

**Comment by t3rminalx** [comment]
> @tomrossi7 In the meantime, I created a fork here with the user agent fix: https://github.com/wiseindy/podgrab
> 
> If you use docker and want to use it, you can swap out the image `akhilrex/podgrab` with `wiseindy/podgrab` until we wait for the PR to be merged in this repo.

It's working like a charm :-)

**Comment by neontechtra** [comment]
Just commenting to follow along, having some issues here as well.

**Comment by t3rminalx** [comment]
So the docker image from @wiseindy and the updated image from @akhilrex both now enable me to add the podcast I was having trouble with however it’s not grabbing the episode artwork or anything newer than when the podcast was initially added.

**Comment by ghenry22** [comment]
Having the same issue with buzzsprout, what should be setting the user agent to here? Something to mimic a browser or something specific to identify podgrab?

**Comment by ghenry22** [comment]
> So the docker image from @wiseindy and the updated image from @akhilrex both now enable me to add the podcast I was having trouble with however it’s not grabbing the episode artwork or anything newer than when the podcast was initially added.

what did you use for the user-agent setting?  I've tried leaving it blank or setting it the same as chrome on my mac and in either case still get the EOF error with a 403 denied in the container logs.  using latest akhilrex docker container from docker hub

**Comment by tomrossi7** [comment]
You should set the user-agent to something to identify yourself or your app -- do not spoof Chrome or any other app since you would then get blocked.

**Comment by ghenry22** [comment]
@tomrossi7 yep no problem, the issue I was seeing was that the user agent was still not being applied at one point in the code so it wasn't working regardless of what I put in there, have submitted a PR to resolve.

Once that gets merged I'll put together something to ensure that a default user-agent that identifies podgrab and the version is used instead of the library default or people putting in random spoofing unnecessarily.

**Comment by tomrossi7** [comment]
Perfect! Thanks!

**Comment by chiefy** [comment]
I'm still seeing this issue even when using @wiseindy 's version `2022.07.07`

**Comment by zanish** [comment]
Is there any update to this? I see PR #243 hasn't been touched in a few months. I'm still seeing the issue.

**Comment by thrillhouse2342** [comment]
I'm unfortunately also experiencing this issue. Just tried setting it up for the first time and the 4 that failed were all buzzsprout.

**Comment by 2600box** [comment]
I had trouble with this and a buzzsprouts podcast. setting the docker image to `wiseindy/podgrab` and setting the user-agent to something `Podgrab - docker` worked

**Comment by Patriot2407** [comment]
I'm also having this issue +1

**Comment by Patriot2407** [comment]
@2600box comment above solved my issue
